### PR TITLE
fix minor issue with endpoint bearertoken

### PR DESCRIPTION
**Pull request by heliapb**
## Description

Fixes issue https://github.com/prometheus-operator/prometheus-operator/issues/5713

## Type of change

_What type of changes does your code introduce to the Prometheus operator? Put an `x` in the box that apply._

- [ ] `CHANGE` (fix or feature that would cause existing functionality to not work as expected)
- [ ] `FEATURE` (non-breaking change which adds functionality)
- [x] `BUGFIX` (non-breaking change which fixes an issue)
- [ ] `ENHANCEMENT` (non-breaking change which improves existing functionality)
- [ ] `NONE` (if none of the other choices apply. Example, tooling, build system, CI, docs, etc.)

## Changelog entry

_Please put a one-line changelog entry below. This will be copied to the changelog file during the release process._

```release-note

```

### CODE REVIEW

**Comment by slashpai** [comment]
@heliapb You need to regenerate after updating type and 
`make generate` command

**Review by ArthurSens — COMMENTED** [review]
I'm struggling to understand why the e2e-tests failed :( 

Locally I'm getting rate limit errors, not sure if that's related

File: pkg/assets/store.go [hunk 1]
```diff
@@ -219,8 +219,8 @@ func (s *Store) addToken(ctx context.Context, ns string, sel v1.SecretKeySelecto
 	return nil
 }
 
-func (s *Store) AddBearerToken(ctx context.Context, ns string, sel v1.SecretKeySelector, key string) error {
-	err := s.addToken(ctx, ns, sel, key)
+func (s *Store) AddBearerToken(ctx context.Context, ns string, sel *v1.SecretKeySelector, key string) error {
+	err := s.addToken(ctx, ns, *sel, key)
```
[the remark this answers is not in the document]
Comment (heliapb) [reply pkg/assets/store.go]: Hi there, thanks for the feedback, changed as suggested. If you might have any other idea about the e2e failing would be much appreciated, as it is my first attempt at contributing to prometheus operator and I'm still a bit lost.

File: pkg/prometheus/resource_selector.go [hunk 2]
```diff
@@ -125,7 +125,7 @@ func (rs *ResourceSelector) SelectServiceMonitors(ctx context.Context, listFn Li
 
 			smKey := fmt.Sprintf("serviceMonitor/%s/%s/%d", sm.GetNamespace(), sm.GetName(), i)
 
-			if err = rs.store.AddBearerToken(ctx, sm.GetNamespace(), endpoint.BearerTokenSecret, smKey); err != nil {
+			if err = rs.store.AddBearerToken(ctx, sm.GetNamespace(), *endpoint.BearerTokenSecret, smKey); err != nil {
```
Comment (simonpasquier) [inline pkg/prometheus/resource_selector.go:128]: I'd rather update the `AddBearerToken()` method to check for nil parameter and return early.

Comment (heliapb) [reply pkg/prometheus/resource_selector.go:128]: Hi @simonpasquier changed, if you could review!

File: pkg/assets/store.go [hunk 3]
```diff
@@ -220,6 +220,10 @@ func (s *Store) addToken(ctx context.Context, ns string, sel v1.SecretKeySelecto
 }
 
 func (s *Store) AddBearerToken(ctx context.Context, ns string, sel v1.SecretKeySelector, key string) error {
+	if sel == (v1.SecretKeySelector{}) {
+		return nil
+	}
+
```
Comment (simonpasquier) [inline pkg/assets/store.go:226]: Sorry for being unclear, I meant to change the signature of AddBearerToken() like this:
```suggestion
func (s *Store) AddBearerToken(ctx context.Context, ns string, sel *v1.SecretKeySelector, key string) error {
	if sel == nil {
		return nil
	}

```

Comment (heliapb) [reply pkg/assets/store.go:226]: Hi if I set it that way I get an error `invalid operation: sel == nil (mismatched types "k8s.io/api/core/v1".SecretKeySelector and untyped nil)compiler[MismatchedTypes](https://pkg.go.dev/golang.org/x/tools/internal/typesinternal#MismatchedTypes)`

Comment (simonpasquier) [reply pkg/assets/store.go:226]: hmm this commit should work https://github.com/simonpasquier/prometheus-operator/commit/a94c876861773f182f5fd3845ede028d9dc40ffc

Comment (heliapb) [reply pkg/assets/store.go:226]: Well in that change you are chaning the func addToken instead of AddBearerToken. So could you please clarify?

Comment (simonpasquier) [reply pkg/assets/store.go:226]: yes I suggest to change both function signatures.

Comment (heliapb) [reply pkg/assets/store.go]: Hi @simonpasquier done, when you have the time pls review

**Review by ArthurSens — APPROVED** [review]
LGTM, thanks for the effort!

**Pull request merged**